rasa/shared/core/domain.py

```diff
@@ -1068,15 +1068,90 @@ def get_active_states(self, tracker: "DialogueStateTracker") -> State:
         }
         return self._clean_state(state)
 
+    @staticmethod
+    def _remove_rule_only_features(
+        state: State, tracker: "DialogueStateTracker"
+    ) -> None:
+        # remove only rule slots
```

Comment:
let's be a bit verbose here so that it gets easier to understand 
```suggestion
        # remove slots which only occur in rules but not in stories
```